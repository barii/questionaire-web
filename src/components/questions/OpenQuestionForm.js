import React from 'react';
import TextFieldGroup from '../common/TextFieldGroup';
import { Form, Button, Card, CardHeader, CardBody } from 'reactstrap';


export default function OpenQuestionForm({question, errors, onChange, onSubmit}) {
  return (
    <div>
    <Card>
      <CardHeader>Open Question</CardHeader>
      <CardBody>
        <Form onSubmit={onSubmit} >
          <TextFieldGroup
            label="Text"
            name="text"
            value={question.text}
            onChange={(e) => onChange(e)}
            error={errors.text}
          />
          <TextFieldGroup
            label="Description"
            name="description"
            value={question.description}
            onChange={(e) => onChange(e)}
            error={errors.description}
          />

          <Button type="submit" color="success">Save</Button>
        </Form>
      </CardBody>
    </Card>
  </div>
  )
}
