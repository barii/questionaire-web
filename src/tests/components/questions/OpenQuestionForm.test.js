import React from 'react';
import { shallow } from 'enzyme';
import OpenQuestionForm from '../../../components/questions/OpenQuestionForm';

const question = {
  type: "OpenQuestion",
  id: '1',
  text: "haha",
  description: "desc"
}

const onSubmit = jest.fn()
const onChange = jest.fn()

const wrapper = shallow(
  <OpenQuestionForm
    question={question} 
    errors
    onChange={onChange} 
    onSubmit={onSubmit} 
  />
);

it('should render OpenQuestionForm correctly (snapshot)', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should call onSubmit on button click', () => {
  wrapper.find('Form').simulate('submit');
  expect(onSubmit).toHaveBeenCalled();
});

it('should call onChange on value update', () => {
  wrapper.find('TextFieldGroup[name="text"]').simulate('change');
  expect(onChange).toHaveBeenCalled();
});
