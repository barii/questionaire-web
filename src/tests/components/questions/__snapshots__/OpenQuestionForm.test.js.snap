// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render OpenQuestionForm correctly (snapshot) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OpenQuestionForm
    errors={true}
    onChange={[Function]}
    onSubmit={[Function]}
    question={
        Object {
            "description": "desc",
            "id": "1",
            "text": "haha",
            "type": "OpenQuestion",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        tag="div"
>
        <CardHeader
                tag="div"
        >
                Open Question
        </CardHeader>
        <CardBody
                tag="div"
        >
                <Form
                        onSubmit={[Function]}
                        tag="form"
                >
                        <TextFieldGroup
                                error={undefined}
                                label="Text"
                                name="text"
                                onChange={[Function]}
                                type="text"
                                value="haha"
                        />
                        <TextFieldGroup
                                error={undefined}
                                label="Description"
                                name="description"
                                onChange={[Function]}
                                type="text"
                                value="desc"
                        />
                        <Button
                                color="success"
                                tag="button"
                                type="submit"
                        >
                                Save
                        </Button>
                </Form>
        </CardBody>
</Card>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardHeader
            tag="div"
>
            Open Question
</CardHeader>,
          <CardBody
            tag="div"
>
            <Form
                        onSubmit={[Function]}
                        tag="form"
            >
                        <TextFieldGroup
                                    error={undefined}
                                    label="Text"
                                    name="text"
                                    onChange={[Function]}
                                    type="text"
                                    value="haha"
                        />
                        <TextFieldGroup
                                    error={undefined}
                                    label="Description"
                                    name="description"
                                    onChange={[Function]}
                                    type="text"
                                    value="desc"
                        />
                        <Button
                                    color="success"
                                    tag="button"
                                    type="submit"
                        >
                                    Save
                        </Button>
            </Form>
</CardBody>,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Open Question",
            "tag": "div",
          },
          "ref": null,
          "rendered": "Open Question",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Form
              onSubmit={[Function]}
              tag="form"
>
              <TextFieldGroup
                            error={undefined}
                            label="Text"
                            name="text"
                            onChange={[Function]}
                            type="text"
                            value="haha"
              />
              <TextFieldGroup
                            error={undefined}
                            label="Description"
                            name="description"
                            onChange={[Function]}
                            type="text"
                            value="desc"
              />
              <Button
                            color="success"
                            tag="button"
                            type="submit"
              >
                            Save
              </Button>
</Form>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TextFieldGroup
                  error={undefined}
                  label="Text"
                  name="text"
                  onChange={[Function]}
                  type="text"
                  value="haha"
/>,
                <TextFieldGroup
                  error={undefined}
                  label="Description"
                  name="description"
                  onChange={[Function]}
                  type="text"
                  value="desc"
/>,
                <Button
                  color="success"
                  tag="button"
                  type="submit"
>
                  Save
</Button>,
              ],
              "onSubmit": [Function],
              "tag": "form",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": undefined,
                  "label": "Text",
                  "name": "text",
                  "onChange": [Function],
                  "type": "text",
                  "value": "haha",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": undefined,
                  "label": "Description",
                  "name": "description",
                  "onChange": [Function],
                  "type": "text",
                  "value": "desc",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Save",
                  "color": "success",
                  "tag": "button",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Save",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          tag="div"
>
          <CardHeader
                    tag="div"
          >
                    Open Question
          </CardHeader>
          <CardBody
                    tag="div"
          >
                    <Form
                              onSubmit={[Function]}
                              tag="form"
                    >
                              <TextFieldGroup
                                        error={undefined}
                                        label="Text"
                                        name="text"
                                        onChange={[Function]}
                                        type="text"
                                        value="haha"
                              />
                              <TextFieldGroup
                                        error={undefined}
                                        label="Description"
                                        name="description"
                                        onChange={[Function]}
                                        type="text"
                                        value="desc"
                              />
                              <Button
                                        color="success"
                                        tag="button"
                                        type="submit"
                              >
                                        Save
                              </Button>
                    </Form>
          </CardBody>
</Card>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardHeader
              tag="div"
>
              Open Question
</CardHeader>,
            <CardBody
              tag="div"
>
              <Form
                            onSubmit={[Function]}
                            tag="form"
              >
                            <TextFieldGroup
                                          error={undefined}
                                          label="Text"
                                          name="text"
                                          onChange={[Function]}
                                          type="text"
                                          value="haha"
                            />
                            <TextFieldGroup
                                          error={undefined}
                                          label="Description"
                                          name="description"
                                          onChange={[Function]}
                                          type="text"
                                          value="desc"
                            />
                            <Button
                                          color="success"
                                          tag="button"
                                          type="submit"
                            >
                                          Save
                            </Button>
              </Form>
</CardBody>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Open Question",
              "tag": "div",
            },
            "ref": null,
            "rendered": "Open Question",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Form
                onSubmit={[Function]}
                tag="form"
>
                <TextFieldGroup
                                error={undefined}
                                label="Text"
                                name="text"
                                onChange={[Function]}
                                type="text"
                                value="haha"
                />
                <TextFieldGroup
                                error={undefined}
                                label="Description"
                                name="description"
                                onChange={[Function]}
                                type="text"
                                value="desc"
                />
                <Button
                                color="success"
                                tag="button"
                                type="submit"
                >
                                Save
                </Button>
</Form>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <TextFieldGroup
                    error={undefined}
                    label="Text"
                    name="text"
                    onChange={[Function]}
                    type="text"
                    value="haha"
/>,
                  <TextFieldGroup
                    error={undefined}
                    label="Description"
                    name="description"
                    onChange={[Function]}
                    type="text"
                    value="desc"
/>,
                  <Button
                    color="success"
                    tag="button"
                    type="submit"
>
                    Save
</Button>,
                ],
                "onSubmit": [Function],
                "tag": "form",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "error": undefined,
                    "label": "Text",
                    "name": "text",
                    "onChange": [Function],
                    "type": "text",
                    "value": "haha",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "error": undefined,
                    "label": "Description",
                    "name": "description",
                    "onChange": [Function],
                    "type": "text",
                    "value": "desc",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Save",
                    "color": "success",
                    "tag": "button",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Save",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
